source/Halibut/HalibutRuntimeBuilder.cs

```diff
@@ -74,6 +75,12 @@ internal HalibutRuntimeBuilder WithPollingReconnectRetryPolicy(Func<RetryPolicy>
             return this;
         }
 
+        public HalibutRuntimeBuilder WithOnUnauthorizedClientConnect(Func<string, string, UnauthorizedClientConnectResponse> onUnauthorizedClientConnect)
```

Comment:
This isn't needed, but it feels like we should start pushing OnUnauthorizedClientConnect to the builder (and eventually remove the public 'set' for this callback)